Satisfactory PHP version comparison fails

## Bug report

| Question    | Answer
| ------------| ---------------
| Box version | Can't work out how to get this, as I'm using a package which I think pulls in a box phar... somewhere - but I've just installed it fresh today
| PHP version | 8.0.12
| Platform with version | MacOS 12.0.1 (tried on Intel & M1)
| Github Repo | - [https://github.com/hotmeteor/eco-cli](hotmeteor/eco-cli)

I'm trying to use Eco CLI, a package which manages Laravel .env values. It's built on Laravel Zero, which I believe uses box.

Running Eco for this first time should be as simple as `eco init` - but this gives:

```bash
Box Requirements Checker
========================

> Using PHP 8.0.12
> PHP is using the following php.ini file:
  /opt/homebrew/etc/php/8.0/php.ini

> Checking Box requirements:
  E................


 [ERROR] Your system is not ready to run the application.


Fix the following mandatory requirements:
=========================================

 * The application requires the version "^7.3" or greater.
```

It looks to me like Box is correctly identifying PHP v8.0.12. 

Looking through the composer lock file, it looks like all the packages which require old versions are either both v7 and v8 (eg `^7.0 || 8.0`) or v7+ (eg `>=7.0`) (see below). When the warning says `version "^7.3" or greater`, the word 'greater' is confusing - does that mean `^7.3` or `>=7.3`?

I've read the [Requirements Checker](https://github.com/box-project/box/blob/master/doc/requirement-checker.md#requirements-checker) documentation and have managed to get Eco working (to some extent) by using the `BOX_REQUIREMENT_CHECKER=0` setting - but from that point I have further issues with Eco which I can't for sure say aren't related to this.

In terms of the chain of dependency here - particularly where Box is required and installed - I'm out of my depth.

### Steps to reproduce
```bash
mkdir eco-test && cd eco-test
composer require hotmeteor/eco-cli
./vendor/bin/eco init
```

### List of PHP version requirements from `composer.lock`
<details>
 <summary>I parsed out a full list of version requirements, sorted & de-duplicated:</summary>

```
>=5.3
>=5.3.0
>=5.3.3
>=5.4
>=5.5
>=5.6
>=7.0
>=7.0.0
>=7.1
>=7.2
>=7.2.0
>=7.2.5
>=7.3
^5.5.9 || ^7.0 || ^8.0
^7.0 || ^8.0
^7.1 || ^8.0
^7.1.3 || ^8.0
^7.1.8 || ^8.0
^7.2 || ^8.0
^7.2.5 || ^8.0
^7.2.5|^8.0
^7.2|^8.0
^7.3 || ^8
^7.3 || ^8.0
^7.3 || ~8.0.0 || ~8.1.0
^7.3|^8.0
^7.4|^8.0
~7.4.0 || ~8.0.0 || ~8.1.0
```

</details>

--- a/requirement-checker/fixtures/fail-complete/expected-boxmin-stderr.dist
+++ b/requirement-checker/fixtures/fail-complete/expected-boxmin-stderr.dist
@@ -7,7 +7,7 @@ Box Requirements Checker
   WARNING: No configuration file (php.ini) used by PHP!
 
 > Checking Box requirements:
-  ✘ The application requires a version matching "^10".'
+  ✘ The application requires a version matching "^10".
   ✔ The application requires the extension "phar".
   ✔ The package "package-with-extensions" requires the extension "json".
   ✘ The package "package-with-extensions" requires the extension "ldap". Enable
@@ -22,7 +22,7 @@ Enable it or install a polyfill.
 Fix the following mandatory requirements:
 =========================================
 
- * The application requires a version matching "^10".'
+ * The application requires a version matching "^10".
  * The package "package-with-extensions" requires the extension "ldap". Enable
    it or install a polyfill.
  * The package "package-with-extensions" requires the extension "random". Enable

--- a/requirement-checker/fixtures/fail-complete/expected-boxmin-stderr.dist
+++ b/requirement-checker/fixtures/fail-complete/expected-boxmin-stderr.dist
@@ -7,7 +7,7 @@ Box Requirements Checker
   WARNING: No configuration file (php.ini) used by PHP!
 
 > Checking Box requirements:
-  ✘ The application requires the version "^10" or greater.
+  ✘ The application requires a version matching "^10".'
   ✔ The application requires the extension "phar".
   ✔ The package "package-with-extensions" requires the extension "json".
   ✘ The package "package-with-extensions" requires the extension "ldap". Enable
@@ -22,7 +22,7 @@ Enable it or install a polyfill.
 Fix the following mandatory requirements:
 =========================================
 
- * The application requires the version "^10" or greater.
+ * The application requires a version matching "^10".'
  * The package "package-with-extensions" requires the extension "ldap". Enable
    it or install a polyfill.
  * The package "package-with-extensions" requires the extension "random". Enable

--- a/requirement-checker/fixtures/fail-complete/expected-composermin-stderr.dist
+++ b/requirement-checker/fixtures/fail-complete/expected-composermin-stderr.dist
@@ -7,7 +7,7 @@ Box Requirements Checker
   WARNING: No configuration file (php.ini) used by PHP!
 
 > Checking Box requirements:
-  ✘ The application requires a version matching "^10".'
+  ✘ The application requires a version matching "^10".
   ✔ The application requires the extension "phar".
   ✔ The package "package-with-extensions" requires the extension "json".
   ✘ The package "package-with-extensions" requires the extension "ldap". Enable
@@ -22,7 +22,7 @@ Enable it or install a polyfill.
 Fix the following mandatory requirements:
 =========================================
 
- * The application requires a version matching "^10".'
+ * The application requires a version matching "^10".
  * The package "package-with-extensions" requires the extension "ldap". Enable
    it or install a polyfill.
  * The package "package-with-extensions" requires the extension "random". Enable

--- a/requirement-checker/fixtures/fail-complete/expected-composermin-stderr.dist
+++ b/requirement-checker/fixtures/fail-complete/expected-composermin-stderr.dist
@@ -7,7 +7,7 @@ Box Requirements Checker
   WARNING: No configuration file (php.ini) used by PHP!
 
 > Checking Box requirements:
-  ✘ The application requires the version "^10" or greater.
+  ✘ The application requires a version matching "^10".'
   ✔ The application requires the extension "phar".
   ✔ The package "package-with-extensions" requires the extension "json".
   ✘ The package "package-with-extensions" requires the extension "ldap". Enable
@@ -22,7 +22,7 @@ Enable it or install a polyfill.
 Fix the following mandatory requirements:
 =========================================
 
- * The application requires the version "^10" or greater.
+ * The application requires a version matching "^10".'
  * The package "package-with-extensions" requires the extension "ldap". Enable
    it or install a polyfill.
  * The package "package-with-extensions" requires the extension "random". Enable

--- a/requirement-checker/fixtures/pass-complete/expected-boxmin-stderr.dist
+++ b/requirement-checker/fixtures/pass-complete/expected-boxmin-stderr.dist
@@ -7,7 +7,7 @@ Box Requirements Checker
   WARNING: No configuration file (php.ini) used by PHP!
 
 > Checking Box requirements:
-  ✔ The application requires the version ">=5.3" or greater.
+  ✔ The application requires a version matching ">=5.3".'
   ✔ The application requires the extension "phar".
   
                                                                                 

--- a/requirement-checker/fixtures/pass-complete/expected-boxmin-stderr.dist
+++ b/requirement-checker/fixtures/pass-complete/expected-boxmin-stderr.dist
@@ -7,7 +7,7 @@ Box Requirements Checker
   WARNING: No configuration file (php.ini) used by PHP!
 
 > Checking Box requirements:
-  ✔ The application requires a version matching ">=5.3".'
+  ✔ The application requires a version matching ">=5.3".
   ✔ The application requires the extension "phar".
   
                                                                                 

--- a/requirement-checker/fixtures/pass-complete/expected-composermin-stderr.dist
+++ b/requirement-checker/fixtures/pass-complete/expected-composermin-stderr.dist
@@ -7,7 +7,7 @@ Box Requirements Checker
   WARNING: No configuration file (php.ini) used by PHP!
 
 > Checking Box requirements:
-  ✔ The application requires the version ">=5.3" or greater.
+  ✔ The application requires a version matching ">=5.3".'
   ✔ The application requires the extension "phar".
   
                                                                                 

--- a/requirement-checker/fixtures/pass-complete/expected-composermin-stderr.dist
+++ b/requirement-checker/fixtures/pass-complete/expected-composermin-stderr.dist
@@ -7,7 +7,7 @@ Box Requirements Checker
   WARNING: No configuration file (php.ini) used by PHP!
 
 > Checking Box requirements:
-  ✔ The application requires a version matching ">=5.3".'
+  ✔ The application requires a version matching ">=5.3".
   ✔ The application requires the extension "phar".
   
                                                                                 

--- a/src/RequirementChecker/AppRequirementsFactory.php
+++ b/src/RequirementChecker/AppRequirementsFactory.php
@@ -258,21 +258,21 @@ final class AppRequirementsFactory
             'condition' => $requiredPhpVersion,
             'message' => null === $packageName
                 ? sprintf(
-                    'The application requires the version "%s" or greater.',
+                    'The application requires a version matching "%s".',
                     $requiredPhpVersion,
                 )
                 : sprintf(
-                    'The package "%s" requires the version "%s" or greater.',
+                    'The package "%s" requires a version matching "%s".',
                     $packageName,
                     $requiredPhpVersion,
                 ),
             'helpMessage' => null === $packageName
                 ? sprintf(
-                    'The application requires the version "%s" or greater.',
+                    'The application requires a version matching "%s".',
                     $requiredPhpVersion,
                 )
                 : sprintf(
-                    'The package "%s" requires the version "%s" or greater.',
+                    'The package "%s" requires a version matching "%s".',
                     $packageName,
                     $requiredPhpVersion,
                 ),

--- a/tests/RequirementChecker/AppRequirementsFactoryTest.php
+++ b/tests/RequirementChecker/AppRequirementsFactoryTest.php
@@ -168,8 +168,8 @@ class AppRequirementsFactoryTest extends TestCase
                 [
                     'type' => 'php',
                     'condition' => '^7.1',
-                    'message' => 'The application requires the version "^7.1" or greater.',
-                    'helpMessage' => 'The application requires the version "^7.1" or greater.',
+                    'message' => 'The application requires a version matching "^7.1".',
+                    'helpMessage' => 'The application requires a version matching "^7.1".',
                 ],
                 [
                     'type' => 'extension',
@@ -196,8 +196,8 @@ class AppRequirementsFactoryTest extends TestCase
                 [
                     'type' => 'php',
                     'condition' => '^7.1',
-                    'message' => 'The application requires the version "^7.1" or greater.',
-                    'helpMessage' => 'The application requires the version "^7.1" or greater.',
+                    'message' => 'The application requires a version matching "^7.1".',
+                    'helpMessage' => 'The application requires a version matching "^7.1".',
                 ],
                 [
                     'type' => 'extension',
@@ -233,8 +233,8 @@ class AppRequirementsFactoryTest extends TestCase
                 [
                     'type' => 'php',
                     'condition' => '^7.1',
-                    'message' => 'The application requires the version "^7.1" or greater.',
-                    'helpMessage' => 'The application requires the version "^7.1" or greater.',
+                    'message' => 'The application requires a version matching "^7.1".',
+                    'helpMessage' => 'The application requires a version matching "^7.1".',
                 ],
                 [
                     'type' => 'extension',
@@ -261,8 +261,8 @@ class AppRequirementsFactoryTest extends TestCase
                 [
                     'type' => 'php',
                     'condition' => '^7.1',
-                    'message' => 'The application requires the version "^7.1" or greater.',
-                    'helpMessage' => 'The application requires the version "^7.1" or greater.',
+                    'message' => 'The application requires a version matching "^7.1".',
+                    'helpMessage' => 'The application requires a version matching "^7.1".',
                 ],
                 [
                     'type' => 'extension',
@@ -295,8 +295,8 @@ class AppRequirementsFactoryTest extends TestCase
                 [
                     'type' => 'php',
                     'condition' => '^7.1',
-                    'message' => 'The application requires the version "^7.1" or greater.',
-                    'helpMessage' => 'The application requires the version "^7.1" or greater.',
+                    'message' => 'The application requires a version matching "^7.1".',
+                    'helpMessage' => 'The application requires a version matching "^7.1".',
                 ],
                 [
                     'type' => 'extension',
@@ -337,8 +337,8 @@ class AppRequirementsFactoryTest extends TestCase
                 [
                     'type' => 'php',
                     'condition' => '^7.1',
-                    'message' => 'The application requires the version "^7.1" or greater.',
-                    'helpMessage' => 'The application requires the version "^7.1" or greater.',
+                    'message' => 'The application requires a version matching "^7.1".',
+                    'helpMessage' => 'The application requires a version matching "^7.1".',
                 ],
                 [
                     'type' => 'extension',
@@ -501,8 +501,8 @@ class AppRequirementsFactoryTest extends TestCase
                 [
                     'type' => 'php',
                     'condition' => '>=5.3',
-                    'message' => 'The application requires the version ">=5.3" or greater.',
-                    'helpMessage' => 'The application requires the version ">=5.3" or greater.',
+                    'message' => 'The application requires a version matching ">=5.3".',
+                    'helpMessage' => 'The application requires a version matching ">=5.3".',
                 ],
                 [
                     'type' => 'extension',
@@ -574,14 +574,14 @@ class AppRequirementsFactoryTest extends TestCase
                 [
                     'type' => 'php',
                     'condition' => '>=5.3',
-                    'message' => 'The package "beberlei/assert" requires the version ">=5.3" or greater.',
-                    'helpMessage' => 'The package "beberlei/assert" requires the version ">=5.3" or greater.',
+                    'message' => 'The package "beberlei/assert" requires a version matching ">=5.3".',
+                    'helpMessage' => 'The package "beberlei/assert" requires a version matching ">=5.3".',
                 ],
                 [
                     'type' => 'php',
                     'condition' => '^5.3.2 || ^7.0',
-                    'message' => 'The package "composer/ca-bundle" requires the version "^5.3.2 || ^7.0" or greater.',
-                    'helpMessage' => 'The package "composer/ca-bundle" requires the version "^5.3.2 || ^7.0" or greater.',
+                    'message' => 'The package "composer/ca-bundle" requires a version matching "^5.3.2 || ^7.0".',
+                    'helpMessage' => 'The package "composer/ca-bundle" requires a version matching "^5.3.2 || ^7.0".',
                 ],
                 [
                     'type' => 'extension',
@@ -664,14 +664,14 @@ class AppRequirementsFactoryTest extends TestCase
                 [
                     'type' => 'php',
                     'condition' => '>=5.3',
-                    'message' => 'The package "beberlei/assert" requires the version ">=5.3" or greater.',
-                    'helpMessage' => 'The package "beberlei/assert" requires the version ">=5.3" or greater.',
+                    'message' => 'The package "beberlei/assert" requires a version matching ">=5.3".',
+                    'helpMessage' => 'The package "beberlei/assert" requires a version matching ">=5.3".',
                 ],
                 [
                     'type' => 'php',
                     'condition' => '^5.3.2 || ^7.0',
-                    'message' => 'The package "composer/ca-bundle" requires the version "^5.3.2 || ^7.0" or greater.',
-                    'helpMessage' => 'The package "composer/ca-bundle" requires the version "^5.3.2 || ^7.0" or greater.',
+                    'message' => 'The package "composer/ca-bundle" requires a version matching "^5.3.2 || ^7.0".',
+                    'helpMessage' => 'The package "composer/ca-bundle" requires a version matching "^5.3.2 || ^7.0".',
                 ],
                 [
                     'type' => 'extension',
@@ -833,8 +833,8 @@ class AppRequirementsFactoryTest extends TestCase
                 [
                     'type' => 'php',
                     'condition' => '^7.3',
-                    'message' => 'The application requires the version "^7.3" or greater.',
-                    'helpMessage' => 'The application requires the version "^7.3" or greater.',
+                    'message' => 'The application requires a version matching "^7.3".',
+                    'helpMessage' => 'The application requires a version matching "^7.3".',
                 ],
                 [
                     'type' => 'extension',
@@ -895,8 +895,8 @@ class AppRequirementsFactoryTest extends TestCase
                 [
                     'type' => 'php',
                     'condition' => '^7.3',
-                    'message' => 'The application requires the version "^7.3" or greater.',
-                    'helpMessage' => 'The application requires the version "^7.3" or greater.',
+                    'message' => 'The application requires a version matching "^7.3".',
+                    'helpMessage' => 'The application requires a version matching "^7.3".',
                 ],
                 [
                     'type' => 'extension',
@@ -965,8 +965,8 @@ class AppRequirementsFactoryTest extends TestCase
                 [
                     'type' => 'php',
                     'condition' => '^7.3',
-                    'message' => 'The application requires the version "^7.3" or greater.',
-                    'helpMessage' => 'The application requires the version "^7.3" or greater.',
+                    'message' => 'The application requires a version matching "^7.3".',
+                    'helpMessage' => 'The application requires a version matching "^7.3".',
                 ],
                 [
                     'type' => 'extension',

--- a/tests/RequirementChecker/RequirementsDumperTest.php
+++ b/tests/RequirementChecker/RequirementsDumperTest.php
@@ -129,8 +129,8 @@ class RequirementsDumperTest extends TestCase
                   array (
                     'type' => 'php',
                     'condition' => '^7.4',
-                    'message' => 'The package "acme/foo" requires the version "^7.4" or greater.',
-                    'helpMessage' => 'The package "acme/foo" requires the version "^7.4" or greater.',
+                    'message' => 'The package "acme/foo" requires a version matching "^7.4".',
+                    'helpMessage' => 'The package "acme/foo" requires a version matching "^7.4".',
                   ),
                   1 =>
                   array (

